localads/localads_tests/campaign_serialization_test.cpp

```diff
@@ -0,0 +1,47 @@
+#include "testing/testing.hpp"
+
+#include "localads/campaign_serialization.hpp"
+
+#include <random>
+#include <vector>
+
+
+using namespace localads;
+
+namespace
+{
+bool TestSerialization(std::vector<Campaign> const & cs)
+{
+  auto const bytes = Serialize(cs);
+  LOG(LINFO, ("Src msg size:", (8+2+1) * cs.size(),
+              "Comporessed size:", bytes.size()));
+  return cs == Deserialize(bytes);
+}
+
+std::vector<Campaign> GenerateRandomCampaigns(size_t number)
+{
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<> featureIds(1, 600000);
```

Comment:
Please, use numeric_limits<decltype(Campaign::m_featureId)>::max(). The same for other fields.